assets/dev/js/editor/elements/views/container.js

```diff
@@ -468,6 +488,73 @@ const ContainerView = BaseElementView.extend( {
 			this.$el.html5Droppable( this.getDroppableOptions() );
 		}
 	},
+
+	handleGridEmptyView() {
+		const currentContainer = this.getCorrectContainerElement();
+
+		this.moveElementToLastChild(
+			currentContainer,
+			currentContainer.find( '> .elementor-empty-view' ),
+		);
+	},
+
+	moveElementToLastChild( parentWrapperElement, childElementToMove ) {
+		let parent = parentWrapperElement.get( 0 ),
+			child = childElementToMove.get( 0 );
+
+		if ( ! parent || ! child ) {
+			return;
+		}
+
+		if ( parent.lastChild === child ) {
+			return;
+		}
+
+		parent.appendChild( child );
+	},
+
+	getCorrectContainerElement() {
+		return this.isBoxedWidth()
+			? this.$el.find( '> .e-con-inner' )
+			: this.$el;
+	},
+
+	shouldIncrementIndex( side ) {
+		if ( this.isFlexContainer() && this.draggingOnBottomOrRightSide( side ) ) {
+			return true;
+		}
+
+		return this.draggingOnBottomOrRightSide( side ) &&
+			! ( this.isGridContainer() && this.emptyViewIsCurrentlyBeingDraggedOver() );
+	},
+
+	draggingOnBottomOrRightSide( side ) {
+		return [ 'bottom', 'right' ].includes( side );
+	},
+
+	isGridContainer() {
+		return 'grid' === this.getContainer().settings.get( 'container_type' );
+	},
+
+	isFlexContainer() {
+		return 'flex' === this.getContainer().settings.get( 'container_type' );
+	},
+
+	isBoxedWidth() {
+		return 'boxed' === this.getContainer().settings.get( 'content_width' );
+	},
+
+	emptyViewIsCurrentlyBeingDraggedOver() {
+		return this.getCorrectContainerElement().find( '> .elementor-empty-view > .elementor-first-add.elementor-html5dnd-current-element' ).length > 0;
+	},
+
+	reInitEmptyView() {
+		if ( ! this.getCorrectContainerElement().find( '> .elementor-empty-view' ).length ) {
```

Comment:
@udidol This has nothing to do with if the container wrapper is empty, because the empty view needs to show even when the container is not empty. This check is used to ensure that we do not run `showEmptyView()` if there is already an empty view element in the container. This is needed for cases when a control change causes a re-render.

Reply:
Got it